### PR TITLE
Fixes flaky test on tenant_tokens_test.dart

### LINKED ISSUE #246: Fix flaky test on the tenant_tokens_test.dart
This happens on both version 2.14 and 2.19 CI's

image

### PR DESCRIPTION
**Pull request by ahmednfwela**
Fixes #246 

The problem is that `expect` is a void that completes instantly without waiting for the called async function to complete.
this means that all the expect tests will process in parallel instantly, which means the `teardown` function might get called before they finish, leading to `my_index` getting deleted while the tests haven't concluded.

### CODE REVIEW

File: test/tenant_token_test.dart [hunk 1]
```diff
@@ -56,15 +56,16 @@ void main() {
...
-              returnsNormally);
-        }
+          client: client,
+        );
+
+        await Future.wait(
+          possibleRules.map((rule) {
+            final token = admClient.generateTenantToken(admKey.uid!, rule);
+            final custom = MeiliSearchClient(testServer, token);
+            return custom.index('my_index').search('');
```
Comment (brunoocasali) [inline test/tenant_token_test.dart:66]: I read the explanation in the description, but don't we need the expect here? How will the `test` know there is a failure?

Comment (ahmednfwela) [reply test/tenant_token_test.dart:66]: the test will fail if there is an exception thrown anyway

Comment (ahmednfwela) [reply test/tenant_token_test.dart:66]: all `expect` is doing is throw an exception if the matcher isn't met

**Review by brunoocasali — APPROVED** [review]
🎉 🎉 
bors merge

**Pull request merged**